### PR TITLE
MET-11: Define domain layer

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Ok
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | Ok
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    private $standardUnitCode;
+
+    /** @var array */
+    private $units;
+
+    private function __construct(MeasurementFamilyCode $code, UnitCode $standardUnitCode, array $units)
+    {
+        Assert::allIsInstanceOf($units, Unit::class);
+        Assert::minCount($units, 1);
+        $this->assertStandardUnitIsAlsoAUnit($standardUnitCode, $units);
```
Comment (ValentinMumble) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php]:
```suggestion
        $this->assertStandardUnitExists($standardUnitCode, $units);
```

File: src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    private function assertStandardUnitIsAlsoAUnit(UnitCode $standardUnitCode, array $units): void
+    {
+        $isStandardUnitCodePresentInUnits = !empty(
+        array_filter(
+            $units,
+            function (Unit $unit) use ($standardUnitCode) {
+                return $standardUnitCode->equals($unit->code());
+            }
+        )
+        );
```
Comment (ValentinMumble) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php]: Weird indent

File: src/Akeneo/Tool/Bundle/MeasureBundle/Provider/LegacyMeasurementProvider.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+namespace Akeneo\Tool\Bundle\MeasureBundle\Provider;
+
+use Akeneo\Tool\Bundle\MeasureBundle\Model\MeasurementFamily;
+use Akeneo\Tool\Bundle\MeasureBundle\Persistence\MeasurementFamilyRepositoryInterface;
+
+/**
+ * @author    Valentin Dijkstra <valentin.dijkstra@akeneo.com>
+ * @copyright 2020 Akeneo SAS (https://www.akeneo.com)
+ */
+class LegacyMeasurementProvider
```
Comment (tseho) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Provider/LegacyMeasurementProvider.php]: Should we use an interface for this service ? The class is directly referenced in the DI of the other services...

**Comment by SamirBoulil** [comment]
Reopened here https://github.com/akeneo/pim-community-dev/pull/11636

**Pull request closed without merging**